x-pack/plugin/mapper-constant-keyword/src/main/java/org/elasticsearch/xpack/constantkeyword/mapper/ConstantKeywordFieldMapper.java

```diff
@@ -140,6 +144,20 @@ public ValueFetcher valueFetcher(SearchExecutionContext context, String format)
                 ? lookup -> List.of()
                 : lookup -> List.of(value);
         }
+        
+        
+
+        @Override
+        public TermsEnum getTerms(boolean caseInsensitive, String string, SearchExecutionContext queryShardContext) throws IOException {
+            boolean matches = caseInsensitive ? 
+                value.toLowerCase(Locale.ROOT).startsWith(string.toLowerCase(Locale.ROOT)) : 
+                value.startsWith(string);
+            if (matches == false) {
+                return null;
+            }
+            int docCount = queryShardContext.searcher().getIndexReader().numDocs();
```

Comment:
should we use `maxDoc()` to be consistent with keyword fields, which don't ignore deletes?